packages/back-end/src/routers/experiment-template/template.controller.ts

```diff
@@ -0,0 +1,143 @@
+import type { Response } from "express";
+import { z } from "zod";
+import { getContextFromReq } from "back-end/src/services/organizations";
+import { AuthRequest } from "back-end/src/types/AuthRequest";
+import { EventUserForResponseLocals } from "back-end/src/events/event-types";
+import { PrivateApiErrorResponse } from "back-end/types/api";
+import {
+  createTemplateValidator,
+  ExperimentTemplateInterface,
+  UpdateTemplateProps,
+} from "./template.validators";
+
+export const getTemplates = async (
+  req: AuthRequest<unknown, unknown, { project?: string }>,
+  res: Response<{ status: 200; templates: ExperimentTemplateInterface[] }>
+) => {
+  const context = getContextFromReq(req);
+  const { project } = req.query;
+
+  const templates = project
+    ? await context.models.experimentTemplates.getByProject(project)
+    : await context.models.experimentTemplates.getAll();
+
+  const filteredTemplates = templates.filter((t) => {
+    return context.permissions.canReadSingleProjectResource(t.project);
+  });
+
+  res.status(200).json({
+    status: 200,
+    templates: filteredTemplates,
+  });
+};
+
+export type CreateTemplateProps = z.infer<typeof createTemplateValidator>;
+
+// region POST /templates
+
+type CreateTemplateResponse = {
+  status: 200;
+  template: ExperimentTemplateInterface;
+};
+
+/**
+ * POST /Templates
+ * Create a Template resource
+ * @param req
+ * @param res
+ */
+export const postTemplate = async (
+  req: AuthRequest<CreateTemplateProps>,
+  res: Response<
+    CreateTemplateResponse | PrivateApiErrorResponse,
+    EventUserForResponseLocals
+  >
+) => {
+  const context = getContextFromReq(req);
+  const { userId } = context;
+  const template = req.body;
+
```

Comment:
Should we check that the org has the premium feature here and return an error if they don't?